compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchRegionHeuristic.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_DIALECT_FLOW_TRANSFORMS_DISPATCHREGIONHEURISTIC_H_
+#define IREE_COMPILER_DIALECT_FLOW_TRANSFORMS_DISPATCHREGIONHEURISTIC_H_
+
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/SmallVector.h"
+#include "mlir/IR/FunctionInterfaces.h"
+
+namespace mlir {
+class DominanceInfo;
+class Operation;
+
+namespace iree_compiler {
+namespace IREE {
+namespace Flow {
+
+/// Mapping of root op to other ops that are in the same fusion group.
+using FusionGroupMapping =
```

Comment:
Such data structures are too heavy weight. This is effectively a duplication of the entire program, i.e. a separate data structure that is holding all the heavy weight operations of the entire program. I am not sure we should expose things this way.

Reply:
What do you mean by "heavy weight"? Note, it's just storing pointers. It does not have to be exposed publicly. (It can be useful though, if you want to make the heuristic exchangeable in the future.)